LibreHardwareMonitorLib/Interop/Kernel32.cs

```diff
@@ -728,7 +727,7 @@ internal struct SCSI_PASS_THROUGH_WITH_BUFFERS
             [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
             public byte[] SenseBuf;
 
-            [MarshalAs(UnmanagedType.ByValArray, SizeConst = SCSI_PASS_THROUGH_BUFFER_SIZE)]
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4096)]
```

Comment:
If this is always the size of NVME_IDENTIFY_CONTROLLER_DATA, perhaps it would be useful to create a const NVME_IDENTIFY_CONTROLLER_DATA_SIZE so it's obvious why this is this size.

Reply:
From what i've read you can set the size of the buffer of `SCSI_PASS_THROUGH_WITH_BUFFERS` to what ever you need. I've just set it to the size of `NVME_IDENTIFY_CONTROLLER_DATA` to fix this bug, but it can be bigger if needed in the future. Also a const would have to be updated if some day `NVME_IDENTIFY_CONTROLLER_DATA` struct is changed, so I think just using `Marshal.SizeOf` is safer.

And the reason I removed `SCSI_PASS_THROUGH_BUFFER_SIZE` const is to match the code in NVMeIntel/NVMeIntelRst ([1](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/61454e1ae2f2ce18550bbb337719e39a3058884f/LibreHardwareMonitorLib/Interop/Kernel32.cs#L695), [2](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/61454e1ae2f2ce18550bbb337719e39a3058884f/LibreHardwareMonitorLib/Hardware/Storage/NVMeIntel.cs#L42)) and NVMeWindows ([1](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/61454e1ae2f2ce18550bbb337719e39a3058884f/LibreHardwareMonitorLib/Interop/Kernel32.cs#L1248), [2](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/61454e1ae2f2ce18550bbb337719e39a3058884f/LibreHardwareMonitorLib/Hardware/Storage/NVMeWindows.cs#L32)). They also have a bigger than needed buffer size of 4096, no constants, and use `(uint)o.DataBuffer.Length` where the size of the buffer is needed.

Reply:
OK, sounds good. Did you test this already? It looks good to me but I don't have a Samsung NVMe to test.

Reply:
The ssds are detected, the sensor values are valid and I have not crashed since this change.

I tested by running a program that just initializes and lists sensors since I could not reproduce this 100% of the time. And it would crash every couple of runs before the change.

Reply:
Still working OK? It looks good to merge but I cannot test this myself.

Reply:
Yes, I guess it would be good for someone else to also test this but I haven't had any exceptions after this change.